cmd/populator-controller/populator-controller.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"flag"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	populator_machinery "github.com/kubev2v/lib-volume-populator/populator-machinery"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	prefix     = "forklift.konveyor.io"
+	mountPath  = "/mnt/"
+	devicePath = "/dev/block"
+)
+
+const (
+	groupName  = "forklift.konveyor.io"
+	apiVersion = "v1beta1"
+	kind       = "OvirtVolumePopulator"
+	resource   = "ovirtvolumepopulators"
+)
+
+func main() {
+	var httpEndpoint, metricsPath, masterURL, kubeconfig, imageName, namespace string
+
+	// Controller args
+	if f := flag.Lookup("kubeconfig"); f != nil {
+		kubeconfig = f.Value.String()
+	} else {
+		flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	}
+	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flag.StringVar(&imageName, "image-name", "", "Image to use for populating")
+	// Metrics args
+	flag.StringVar(&httpEndpoint, "http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled.")
+	flag.StringVar(&metricsPath, "metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	// Other args
+	flag.StringVar(&namespace, "namespace", "konveyor-forklift", "Namespace to deploy controller")
+	flag.Parse()
+
+	var (
+		gk  = schema.GroupKind{Group: groupName, Kind: kind}
+		gvr = schema.GroupVersionResource{Group: groupName, Version: apiVersion, Resource: resource}
+	)
+	imageName = "quay.io/lrotenbe/ovirt-populator:test"
```

Comment:
will need to change into `kubev2v/ovirt-populator:latest`

Reply:
i think that will be wrong. we create this pod as controller and it should (in the future) be for multiple providers using different images. generally thinking - we will have a switch case here, based on the provider(we do need to think however how it will know which to use). or, we do need to think how we provide the image name. this is env var. so it will be wrong (in my opinion) to do it that way.

Reply:
@bennyz correct me if I'm wrong here but we thought about adding the image to use on the CR rather than using an environment variable, right?

Reply:
yes, but it doesn't look like the library can actually handle different populators without us modifying it. It seems like we need to have getPopulatorPodArgs select the image according the CR (assuming it will have an imageName field)
https://github.com/kubernetes-csi/lib-volume-populator/blob/4adb975aba60aa689893d452d33806befb977efd/populator-machinery/controller.go#L540

and only then set the image https://github.com/kubernetes-csi/lib-volume-populator/blob/4adb975aba60aa689893d452d33806befb977efd/populator-machinery/controller.go#L555

Reply:
if it's in the CR, we can use the populator-controller image to take it in provide it to the the library, without modifying the library further.

Reply:
as we talked offline, we may run the `RunController` twice, once for ovirt and once for OSP. currently, we will get the `image-name` for ovirt. it will be part of the deployment.